cpp/src/arrow/array/builder_binary.h

```diff
@@ -463,6 +464,238 @@ class ARROW_EXPORT LargeStringBuilder : public LargeBinaryBuilder {
   std::shared_ptr<DataType> type() const override { return large_utf8(); }
 };
 
+// ----------------------------------------------------------------------
+// BinaryViewBuilder, StringViewBuilder
+//
+// These builders do not support building raw pointer view arrays.
+
+namespace internal {
+
+// We allocate medium-sized memory chunks and accumulate data in those, which
+// may result in some waste if there are many large-ish strings. If a string
+// comes along that does not fit into a block, we allocate a new block and
+// write into that.
+//
+// Later we can implement optimizations to continuing filling underfull blocks
+// after encountering a large string that required allocating a new block.
+class ARROW_EXPORT StringHeapBuilder {
+ public:
+  static constexpr int64_t kDefaultBlocksize = 32 << 10;  // 32KB
```

Comment:
1. 32KB is the default in Velox. I'm happy to change it however.
2. It seems that setting a larger buffer size would be sufficient in most cases to prevent chunking. If necessary I could add `StringHeapBuilder::realloc_instead_of_chunking_`, but removing chunking altogether would hinder follow ups like adding configurable chunk count to random generation

Reply:
Hmm, I see. Perhaps we can keep chunking but with a default very large chunk size, then.